homeassistant/components/comelit/__init__.py

```diff
@@ -0,0 +1,38 @@
+"""Comelit integration."""
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PIN,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import ComelitSerialBridge
+
+PLATFORMS = [Platform.LIGHT]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Comelit platform."""
+    coordinator = ComelitSerialBridge(hass, entry.data[CONF_HOST], entry.data[CONF_PIN])
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
```suggestion
    await coordinator.async_config_entry_first_refresh()    

    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```
We should not set hass data before the first refresh (_taken from https://github.com/home-assistant/core/pull/97028#discussion_r1273340576_)

Reply:
Ok, noted. I think there should be a review of all integrations then as many of them are in the wrong order.